pxtcompiler/emitter/service.ts

```diff
@@ -1755,6 +1755,8 @@ namespace ts.pxtc.service {
                     functionArgument = `(${functionSignature.parameters.map(p => p.name).join(', ')})`;
                 let n = fnName || "fn";
                 if (functionCount++ > 0) n += functionCount;
+                if (isArgument) // forever -> on_forever
```

Comment:
Should this check if it already starts with `on`? Otherwise won't `on_chat` in minecraft be `on_on_chat`?

Reply:
added check